### PR TITLE
Make HAL+JSON strict.

### PR DESCRIPTION
**Pull request by e0ipso**
Continues from #238.

### CODE REVIEW

**Comment by e0ipso** [comment]
Tests are failing because of entity validator. All PRs are blocked until we fix that.

**Comment by amitaibu** [comment]
Can you retry the test -- I fixed that problem couple of hours ago

‫ב-18 בנוב 2014, בשעה 17:24, ‏Mateu Aguiló Bosch notifications@github.com כתב/ה:‬

**Comment by amitaibu** [comment]
@mateu-aguilo-bosch kicked Travis, and tests are green :) I'l review now.

File: plugins/restful/RestfulEntityBase.php [hunk 1]
```diff
@@ -427,6 +432,14 @@ protected function getValueFromResource(EntityMetadataWrapper $wrapper, $propert
       return $wrapper->value(array('identifier' => TRUE));
     }
 
+    if ($public_field_name) {
+      $this->valueMetadata[$host_id][$public_field_name][] = array(
```
Comment (amitaibu) [inline plugins/restful/RestfulEntityBase.php:436]: maybe we should have a `$this->addValueMetaData($host_id, $public_field_name, $values)`

**Comment by amitaibu** [comment]
@mateu-aguilo-bosch @mirzu 
Maybe we should get this in, and tackle the rest of the issues in follow-ups?

**Comment by e0ipso** [comment]
@amitaibu let's do that. So far I can only see the helper method `addValueMetaData`.

**Comment by amitaibu** [comment]
Merged.
https://github.com/Gizra/restful/commit/62b78e3f3626204d3f2db9a6c64a38a587e21db8

**Pull request closed without merging**